tests/test_auth.py

```diff
@@ -0,0 +1,153 @@
+import os
```

Comment:
Really nice test suite! I few comments I have:
- no need for the classes. In my opinion, it makes things less readable. I know we have plenty of them in other modules but let's consider them as legacy :D When outside of a test class you can use fixtures more easily
- I like the idea of not using the "real" token and profiles paths. To go one step further, we could use a "per-test autouse fixture" that creates a path in tmp directory for each test. The problem I see with the current is if the tests are ran in parallel (which is the case in the CI), they might conflicts with each other

Apart from that, thanks for the complete tests!

Reply:
My bad, I was thinking of unittest classes, no just raw classes.
Using basic classes to group things is fine yes (not so opinionated to be honest ^^). The problem is when you inherit from `unittest.TestCase` (which is what we do everywhere). When doing so, the fixtures cannot be used:

```py
import unittest

import pytest


@pytest.fixture
def hello():
    return "Hello World"

def test_something(hello: str):
    assert hello == "Hello World"

class TestSomethingInClass():
    def test_something(self, hello: str):
        assert hello == "Hello World"

class TestSomethingInUnitTestClass(unittest.TestCase):
    def test_something(self, hello: str):
        assert hello == "Hello World"
```

will lead to 

```
(...)

self = <test.TestSomethingInUnitTestClass testMethod=test_something>
method = <bound method TestSomethingInUnitTestClass.test_something of <test.TestSomethingInUnitTestClass testMethod=test_something>>

    def _callTestMethod(self, method):
>       method()
E       TypeError: TestSomethingInUnitTestClass.test_something() missing 1 required positional argument: 'hello'

/usr/lib/python3.10/unittest/case.py:549: TypeError
================================================================================ slowest durations ================================================================================

(9 durations < 0.005s hidden.  Use -vv to show these durations.)
============================================================================= short test summary info =============================================================================
FAILED test.py::TestSomethingInUnitTestClass::test_something - TypeError: TestSomethingInUnitTestClass.test_something() missing 1 required positional argument: 'hello'
=========================================================================== 1 failed, 2 passed in 0.26s ===========================================================================
```

Reply:
There are ways to mitigate unittest classes with pytest fixture but not very satisfying (see https://docs.pytest.org/en/latest/how-to/unittest.html#using-autouse-fixtures-and-accessing-other-fixtures). It works only for autouse and applies to all tests at once.